analyzers/tests/SonarAnalyzer.UnitTest/TestCases/TestAnalysisComparer_1.cs

```diff
@@ -0,0 +1,18 @@
+﻿namespace Tests.Diagnostics
+{
+    public class FirstClass // Noncompliant
+    {
+        async void CorrectNoncompliant() { } // Noncompliant {{Return 'Task' instead.}}
+        //    ^^^^
+
+        async void Unexpected() { }
+
+        async void WrongMessage() { } // Noncompliant {{Wrong message}}
+
+        async void WrongPlace() { } // Noncompliant {{Return 'Task' instead.}}
+        //         ^^^^^^^^^^
```

Comment:
What if it has a wrong position AND wrong length? Let's document it (whatever it does)